transactor/src/main/scala/io/mediachain/transactor/StateMachine.scala

```diff
@@ -73,6 +85,7 @@ object StateMachine {
       def commit(ref: Reference, newchain: Reference, oldchain: Option[Reference]) = {
         val entry = ChainEntry(nextSeqno(), ref, newchain, oldchain)
         publishCommit(entry)
+        blockExtend(entry)
```

Comment:
ordering of these might need to be flipped and this should be transactional as weel